latency/src/bin/z_ping.rs

```diff
@@ -12,112 +12,130 @@
     let scenario = opt.scenario;
     let name = opt.name;
     let interval = opt.interval;
     task::spawn(async move {
-        let mut sub = c_session
-            .subscribe("/test/pong/")
-            .reliable() //Default of the reliability is `best_effort`
-            .await
-            .unwrap();
-
-        // Notify that the subscriber has been created
-        c_barrier.wait().await;
-
         while let Some(sample) = sub.next().await {
-                  let mut payload_reader = sample.value.payload.reader();   
-                  let mut count_bytes = [0u8; 8];
-                  if payload_reader.read_exact(&mut count_bytes){
-                    let count = u64::from_le_bytes(count_bytes);
+            let mut payload_reader = sample.value.payload.reader();
+            let mut count_bytes = [0u8; 8];
+            if payload_reader.read_exact(&mut count_bytes) {
+                let count = u64::from_le_bytes(count_bytes);
 
-                    let instant = c_pending.lock().await.remove(&count).unwrap();
-                    println!(
-                        "zenoh,{},latency.parallel,{},{},{},{},{}",
-                        scenario,
-                        name,
-                        sample.value.payload.len(),
-                        interval,
-                        count,
-                        instant.elapsed().as_micros()
-                    );
-                }  else {
-                    panic!("Fail to fill the buffer");
-                }
+                let instant = c_pending.lock().await.remove(&count).unwrap();
+                println!(
+                    "zenoh,{},latency.parallel,{},{},{},{},{}",
+                    scenario,
+                    name,
+                    sample.value.payload.len(),
+                    interval,
+                    count,
+                    instant.elapsed().as_micros()
+                );
+            } else {
+                panic!("Fail to fill the buffer");
+            }
         }
         panic!("Invalid value!");
     });
 
-    // Wait for the subscriber to be declared
-    barrier.wait().await;
-
     let mut count: u64 = 0;
     loop {
         let count_bytes: [u8; 8] = count.to_le_bytes();
         let mut payload = vec![0u8; opt.payload];
         payload[0..8].copy_from_slice(&count_bytes);
 
         pending.lock().await.insert(count, Instant::now());
-
-        session
-            .put("/test/ping", payload)
-            .congestion_control(CongestionControl::Block)
-            .await
-            .unwrap();
+        if opt.use_expr {
+            session
+                .put(key_expr_num, payload)
+                .congestion_control(CongestionControl::Block)
+                .await
+                .unwrap();
+        } else {
+            session
+                .put("/test/ping", payload)
+                .congestion_control(CongestionControl::Block)
+                .await
+                .unwrap();
+        }
```

Comment:
IMO, the `z_ping` case is differ to the case of `z_put_thr`, because in `z_ping` case, we need to create the payload in each iteration,  hence we need to create the writer inside the loop ([here](https://github.com/william-wyliang/zenoh-perf/blob/665cd2bc32b74d3d8ce0a11c78a9fc54bd40181e/throughput/src/bin/z_put_thr.rs#L98)), I am not sure is it a good idea to put these actions into the loop (includes `declare_publication` and `declare_expr`).

Reply:
`session.put()` already creates a writer, it's just one way to avoid duplicating the whole builder statement. You can consider that and make your own judgement based on the complexity when `declare_publication` and `declare_expr` are also included.